sympy/matrices/matrixbase.py

```diff
@@ -1329,7 +1329,7 @@ def _eval_is_symbolic(self):
     # routines and has a different *args signature.  Make
     # sure the names don't clash by adding `_matrix_` in name.
     def _eval_is_matrix_hermitian(self, simpfunc):
-        herm = lambda i, j: simpfunc(self[i, j] - self[j, i].conjugate()).is_zero
+        herm = lambda i, j: simpfunc(self[i, j] - self[j, i].transpose().conjugate()).is_zero
```

Comment:
In https://github.com/sympy/sympy/pull/27422 we discussed that transpose shouldn't propagate to matrix entries. Therefore this might not work in general, I believe. Instead I propose using adjoint directly.
```suggestion
        herm = lambda i, j: simpfunc(self[i, j] - self[j, i].adjoint()).is_zero
```

Reply:
I agree with this change